packages/yourloops/test/integration/assert/caregivers-list.assert.ts

```diff
@@ -0,0 +1,184 @@
+    email: newCaregiverEmail,
+    type: NotificationType.directInvitation,
+    target: { id: 'target-id' },
+    id: 'my-id'
+  } as Notification])
+  await userEvent.click(inviteButton)
+
+  expect(DirectShareApi.addDirectShare).toHaveBeenCalledWith(patient1Id, newCaregiverEmail)
+  expect(screen.queryByRole('dialog')).not.toBeInTheDocument()
+
+  const inviteSuccessfulSnackbar = screen.getByTestId('alert-snackbar')
+  expect(inviteSuccessfulSnackbar).toHaveTextContent('Invite sent!')
+  await userEvent.click(within(inviteSuccessfulSnackbar).getByTitle('Close'))
+}
+
+export const checkAddCaregiverCancel = async () => {
+  const addCaregiverButton = screen.getByRole('button', { name: 'Add caregiver' })
+  await userEvent.click(addCaregiverButton)
+
+  const addCaregiverDialog = screen.getByRole('dialog')
+  expect(addCaregiverDialog).toBeVisible()
+
+  const cancelButton = within(addCaregiverDialog).getByRole('button', { name: 'Cancel' })
+  expect(cancelButton).toBeVisible()
+  expect(cancelButton).toBeEnabled()
+
+  await userEvent.click(cancelButton)
+
+  expect(screen.queryByRole('dialog')).not.toBeInTheDocument()
+}
+
+export const checkAddCaregiverErrors = async () => {
+  const addCaregiverButton = screen.getByRole('button', { name: 'Add caregiver' })
+  await userEvent.click(addCaregiverButton)
+
+  const addCaregiverDialog = screen.getByRole('dialog')
+  expect(addCaregiverDialog).toBeVisible()
+
+  const inviteButton = within(addCaregiverDialog).getByRole('button', { name: 'Invite' })
+  expect(inviteButton).toBeVisible()
+  expect(inviteButton).toBeDisabled()
+
+  const emailInput = within(addCaregiverDialog).getByRole('textbox', { name: 'Email' })
+  const validEmail = 'valid-email@test.com'
+  await userEvent.type(emailInput, validEmail)
+  expect(inviteButton).toBeEnabled()
+
+  await userEvent.clear(emailInput)
+  expect(inviteButton).toBeDisabled()
+
+  await userEvent.type(emailInput, 'invalid-email')
+  expect(inviteButton).toBeDisabled()
+  expect(within(addCaregiverDialog).getByText('Invalid email address (special characters are not allowed).')).toBeVisible()
+
+  await userEvent.clear(emailInput)
+  await userEvent.type(emailInput, validEmail)
+  expect(inviteButton).toBeEnabled()
+  expect(within(addCaregiverDialog).queryByText('Invalid email address (special characters are not allowed).')).not.toBeInTheDocument()
+
+  await userEvent.clear(emailInput)
+  await userEvent.type(emailInput, mockCaregiverUser.username)
+  expect(inviteButton).toBeDisabled()
+  expect(within(addCaregiverDialog).getByText('You are already sharing your data with this caregiver.')).toBeVisible()
+
+  await userEvent.clear(emailInput)
+  await userEvent.type(emailInput, 'other-patient@mail.com')
+
+  jest.spyOn(DirectShareApi, 'addDirectShare').mockRejectedValueOnce(new Error(PATIENT_CANNOT_BE_ADDED_AS_CAREGIVER_ERROR_MESSAGE))
+  await userEvent.click(inviteButton)
+
+  expect(DirectShareApi.addDirectShare).toHaveBeenCalled()
```

Comment:
<!-- Nitpicks are trivial preference-based requests. These should be non-blocking by nature.  -->
**🤓 nitpick:** ‏You can check with what parameters it has been called :slightly_smiling_face: